src/backend/vulkan/src/device.rs

```diff
@@ -2274,24 +2274,34 @@ impl Device {
     unsafe fn set_object_name(&self, object_type: vk::ObjectType, object_handle: u64, name: &str) {
         let instance = &self.shared.instance;
         if let Some(DebugMessenger::Utils(ref debug_utils_ext, _)) = instance.1 {
-            // Append a null terminator to the string while avoiding allocating memory
-            static mut NAME_BUF: [u8; 64] = [0u8; 64];
-            std::ptr::copy_nonoverlapping(
-                name.as_ptr(),
-                &mut NAME_BUF[0],
-                name.len().min(NAME_BUF.len()),
-            );
-            NAME_BUF[name.len()] = 0;
-            let _result = debug_utils_ext.debug_utils_set_object_name(
-                self.shared.raw.handle(),
-                &vk::DebugUtilsObjectNameInfoEXT {
-                    s_type: vk::StructureType::DEBUG_UTILS_OBJECT_NAME_INFO_EXT,
-                    p_next: std::ptr::null_mut(),
-                    object_type,
-                    object_handle,
-                    p_object_name: NAME_BUF.as_ptr() as *mut _,
-                },
-            );
+            let set_name = |name_ptr| {
+                let _result = debug_utils_ext.debug_utils_set_object_name(
+                    self.shared.raw.handle(),
+                    &vk::DebugUtilsObjectNameInfoEXT {
+                        s_type: vk::StructureType::DEBUG_UTILS_OBJECT_NAME_INFO_EXT,
+                        p_next: std::ptr::null_mut(),
+                        object_type,
+                        object_handle,
+                        p_object_name: name_ptr as *const i8,
+                    },
+                );
+            };
+
+            // Append a null terminator to the string
+            if name.len() < 64 {
```

Comment:
let's do `let name_ptr = if ... else ...;` so that we don't have to have a helper function.
Note that you'd need to move the declaration out of if/else:
```rust
let mut buf_vect: Vec<u8>;
if ... {...} else {
buffer = (name.as_bytes().iter().cloned().chain(std::iter::once(0)))
                    .collec();
    buf_vect.as_mut_ptr()
};
```

Reply:
Yeah that's probably better.